woocommerce/woocommerce-gutenberg-products-block
# Move Block Type Settings into Block Type Classes

Note this PR is based on https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/4055 not trunk. This is part of #3748

**This PR changes how settings are defined for blocks.**

1. **Core Settings.** These are shared/global and useful to blocks but also other projects. Core settings are exported from the `@woocommerce/settings` package.
2. **Block Config Settings.** These are shared between multiple blocks e.g. build properties. They are injected on all pages.
3. **Block Type Settings.** These are defined by block types themselves.

**Block Constants** now only exports the Block Config (build) and derived settings e.g. a page permalink.

Block Type Settings are not defined as constants, and instead use `getSetting` inline. These should not be constant because they will not exist on all requests.

In addition to standardising and reviewing the current settings, this PR:

- Implements TypeScript in the settings packages
- Moves some settings from block to core to block type and vice versa
- Ensures settings are added via AssetDataApi rather than using other methods.

### How to test the changes in this Pull Request:

1. Smoke test product blocks - All Products and SSR blocks - confirm max/min/default rows and columns work as expected
2. Smoke test cart and checkout
3. Tests should pass - I updated globals so that the suite still works.

This PR is large because it swaps constants to getSetting across all files.
